### PR TITLE
CSP-751 CSP-750 Implements Hybrid Event Sign Up 

### PR DESCRIPTION
**Pull request by SawNawDfE**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs [hunk 1]
```diff
@@ -1,29 +1,52 @@
...
 using SFA.DAS.ApprenticeAan.Web.Extensions;
 using SFA.DAS.ApprenticeAan.Web.Infrastructure;
 using SFA.DAS.ApprenticeAan.Web.Models;
+using SFA.DAS.ApprenticeAan.Web.Models.NetworkEvents;
 
 namespace SFA.DAS.ApprenticeAan.Web.Controllers;
 
 [Authorize]
-[Route("network-events", Name = RouteNames.NetworkEvents)]
+[Route("network-events")]
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:13]: this needs reverting, that ', Name = RouteNames.NetworkEvents' is required

Comment (MarkFCain) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:13]: oh I see, ignore this

Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:13]: you need to take th route out of this

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs [hunk 2]
```diff
@@ -1,29 +1,52 @@
...
 
-    public NetworkEventsController(IOuterApiClient outerApiClient)
+    public NetworkEventsController(IOuterApiClient outerApiClient, ApplicationConfiguration applicationConfiguration)
     {
         _outerApiClient = outerApiClient;
+        _applicationConfiguration = applicationConfiguration;
     }
+
     [HttpGet]
+    [Route("", Name = RouteNames.NetworkEvents)]
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:26]: You need to put in Route("network-events", ... here

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs [hunk 3]
```diff
@@ -1,29 +1,52 @@
...
     {
         var calendarEventsResponse = await _outerApiClient.GetCalendarEvents(User.GetAanMemberId(), cancellationToken);
         var model = (NetworkEventsViewModel)calendarEventsResponse;
         return View(model);
     }
-}
\ No newline at end of file
+
+    [HttpGet]
+    [Route("{id}", Name = RouteNames.NetworkEventDetails)]
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:35]: you need to change the Route to "network-events/{id}"

File: src/SFA.DAS.ApprenticeAan.Web/Views/NetworkEvents/Details.cshtml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+@model NetworkEventDetailsViewModel
+
+@{
+    ViewData["Title"] = $"Event Page — {Model.Title}";
+}
+
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        <strong class="govuk-tag app-tag app-tag--@Model.EventFormatAppTagSuffix">@Model.EventFormatAppTagValue</strong> <strong class="govuk-tag app-tag">@Model.CalendarName</strong>
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Views/NetworkEvents/Details.cshtml:12]: I have some things to discuss with you about EventFromatAppTagSuffix and EventFormatAppTagValue.  Let's pair

File: src/SFA.DAS.ApprenticeAan.Web/Models/NetworkEvents/NetworkEventDetailsViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        return EventFormat switch
+        {
+            EventFormat.Online => "_OnlineEventPartial.cshtml",
+            EventFormat.InPerson => "_InPersonEventPartial.cshtml",
+            EventFormat.Hybrid => "_HybridEventPartial.cshtml",
+            _ => throw new NotImplementedException($"Failed to find a matching partial view for event format \"{EventFormat}\""),
+        };
+    }
+
+    private string GetEventFormatAppTagValue()
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Models/NetworkEvents/NetworkEventDetailsViewModel.cs:84]: I want to discuss a different way to do this, as I've done in CSP-731

**Comment by SawNawDfE** [comment]
I've created a new branch off main and applied the changes in here to that. It was easier than the rebase was looking to be. 
https://github.com/SkillsFundingAgency/das-apprentice-aan-web/pull/52

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs (hunk 1 above)
Comment (SawNawDfE) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:13]: Done in new PR -- please see bottom comment

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs (hunk 2 above)
Comment (SawNawDfE) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:26]: Done in new PR -- please see bottom comment

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs (hunk 3 above)
Comment (SawNawDfE) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkEventsController.cs:35]: Done in new PR -- please see bottom comment

**Pull request closed without merging**